UnitySDK/Assets/ML-Agents/Scripts/Brain.cs

```diff
@@ -94,7 +87,18 @@ private void Shutdown()
         /// </summary>
         private void BrainDecideAction()
         {
-            m_BrainBatcher?.SendBrainInfo(name, m_AgentInfos);
+            m_Communicator?.PutObservations(name, m_AgentInfos);
+            var actions = m_Communicator?.GetActions(name);
+            if (actions != null)
+                foreach (var tuple in actions)
+                {
+                    tuple.Item1.UpdateVectorAction(tuple.Item2.vectorActions);
```

Comment:
Can you make it clearer what Item1 and Item2 are here? E.g.
``` csharp
var agent = tuple.Item1;
var action = tuple.Item2;
agent.UpdateMemoriesAction(action.memories);
// etc
```

Reply:
Yeah, I'm actually wondering if we can just have an UpdateAction(AgentAction action) on agent and remove these other methods.  What do you think about that?

Reply:
Removed Tuples